File: dotnet/ContosoPizza/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace ContosoPizza.Models;

public class Order
{
    public int Id { get; set; }
    [Required]
    public int CustomerId { get; set; }
    [Required]
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    [Required]
    public decimal TotalAmount { get; set; }

    public Customer? Customer { get; set; }
    public ICollection<OrderItem>? OrderItems { get; set; } = [];
}

File: angular/ContossoPizza-Frontend/src/app/shared/models/Order.ts
export interface Order {
    id: number;
    customerName?: string;
    orderDate: string;
    totalAmount: number;
}

export interface OrderItem {
    pizzaId: number;
    quantity: number;
}

export interface OrderPayload {
    customerId: number;
    orderItems: OrderItem[];
}
